chapter6.adoc

```diff
@@ -67,24 +67,85 @@ response to the address translation request on the device translation completion
-walk interface (MWI).
+walk interface (MWI). To perform the checks the MTTCHK uses the same MTT table
+format as used by the CPU's MMU. Using the same MTT table formats as the CPU's
+MMU allows the same table to be used simultaneously by the CPU MMU and MTTCHK.
+The MTT access permission lookup process used by MTTCHK is identical to that
+specified by Smmtt extension in section "MTT access permissions lookup process".
+
+The RISC-V memory model requires memory access from a hart to be single-copy
+atomic. When RV32 is implemented the size of a single-copy atomic memory access
+is up to 32-bits. When RV64 is implemented the size of a single-copy atomic
+memory access is up to 64-bits. The size of a single-copy atomic memory access
+implemented by MTTCHK is UNSPECIFIED but is required to be at least 32-bits if
+all of the harts in the system implement RV32 and is required to be at least
+64-bits if any of the harts in the system implement RV64. Software must follow
+the rules outlined below to update MTT entries.
+
+* It is generally unsafe to update fields of an MTT entry using stores of width
+  less than the minimal single-copy atomic memory access supported by MTTCHK as
+  it is legal for MTTCHK to read the entry at any time, including when only
+  some of the partial stores have taken effect. For an update to be atomic,
+  software must use a single store of width equal to the minimal single-copy
+  atomic memory access supported by MTTCHK.
+
+* MTTCHK is not required to immediately observe the software updates to an MTT
+  entry. Software must use the `MTTINVAL` operation outlined in <<section>> to
+  invalidate any previous copies of that entry that may be in the MTTCHK caches
+  to synchronize the updates to the entry with the operation of MTTCHK.
 
-Each IOMMU offers a memory-mapped register programming interface that the
-associated supervisor domain uses to configure and control the IOMMU. The RISC-V
-hart employs the MTT to enforce access controls on the physical address. This is
-done to ascertain whether the supervisor domain currently executing on that
-RISC-V hart possesses the rights to access the physical address of an IOMMU
-register programming interface.
+[NOTE]
+====
+If an MTT entry is changed, MTTCHK may use the old value of the entry or the
+new value of the entry and the choice is unpredictable until software uses the
+`MTTINVAL` operation to synchronize updates to the entry with the operation of
+the MTTCHK. These are the only behaviors expected.
+====
 
-The I/O MTT checker provides a memory-mapped register programming interface
-associated with the RDSM. The RDSM employs the MTT (or PMP) to prohibit access
-to these registers from any of the supervisor domains.
+The I/O MTT checker provides a memory-mapped register programming interface that
+is expected to be associated with the RDSM. The RDSM may employ the MTT (or PMP)
+to prohibit access to these registers from any of the supervisor domains.
```

Comment:
"expected to be associated with" -> expected to be M-mode accessible?

Do we need to say "The RDSM may employ..." - like the other case that is implementation choice (probably a sane one) but no need to specify it?

Reply:
Yes, its repeating the obvious. So I will remove the "that is expected ...." and the sentence that follows it.